tests/integration_tests/saas/test_mailchimp_task.py

```diff
@@ -123,6 +123,87 @@ def test_saas_access_request_task(
     )
 
 
+@pytest.mark.integration_saas
+@pytest.mark.integration_hubspot
+def test_saas_access_request_task(
+        db,
+        policy,
+        connection_config_hubspot,
+        dataset_config_hubspot,
+        hubspot_identity_email,
+) -> None:
+    """Full access request based on the Mailchimp SaaS config"""
+
+    privacy_request = PrivacyRequest(
+        id=f"test_saas_access_request_task_{random.randint(0, 1000)}"
+    )
+    identity_attribute = "email"
+    identity_value = hubspot_identity_email
+    identity_kwargs = {identity_attribute: identity_value}
+    identity = PrivacyRequestIdentity(**identity_kwargs)
+    privacy_request.cache_identity(identity)
+
+    dataset_name = connection_config_hubspot.get_saas_config().fides_key
+    merged_graph = dataset_config_hubspot.get_graph()
+    graph = DatasetGraph(merged_graph)
+
+    v = graph_task.run_access_request(
+        privacy_request,
+        policy,
+        graph,
+        [connection_config_hubspot],
+        {"email": hubspot_identity_email},
+    )
+
+    assert_rows_match(
+        v[f"{dataset_name}:contacts"],
+        min_size=1,
+        keys=[],
+    )
+    assert_rows_match(
+        v[f"{dataset_name}:subscription_preferences"],
+        min_size=1,
+        keys=[],
+    )
+
```

Comment:
Can you adjust the tests so that the keys are not an empty list? I'd like to verify in the test what the row labels are.  


Also, it might be worthwhile filtering the data with a policy with some data_categories at the end of this test so you can high-level verify what results are returned.  See the equivalent sentry test.  As noted elsewhere, it seems like the `subscription_preferences` dataset is incorrect, so you're not going to get data returned there, but it's not obvious from this test.

Reply:
thanks for adding the details to the test!